src/rdkafka.c

```diff
@@ -3181,30 +3202,58 @@ rd_kafka_message_t *rd_kafka_consumer_poll(rd_kafka_t *rk, int timeout_ms) {
 }
 
 
-rd_kafka_resp_err_t rd_kafka_consumer_close(rd_kafka_t *rk) {
+/**
+ * @brief Consumer close.
+ *
+ * @param rebalance_rkq If specified; rebalance events (etc) will be forwarded
+ *                      to this queue which must be served by the application
+ *                      until this call to close_queue() returns.
+ *                      If the consumer is not in a joined state, no rebalance
+ *                      events will be emitted.
+ *                      If not specified; this function will serve the rebalance
+ *                      callbacks.
+ *
+ */
+static rd_kafka_error_t *
+rd_kafka_consumer_close_q(rd_kafka_t *rk, rd_kafka_q_t *rebalance_rkq) {
```

Comment:
i'd consider two separate methods, even though there would be some duplication, though you could maybe factor that out.